sw/device/lib/dif/autogen/dif_otp_ctrl_autogen.h

```diff
@@ -0,0 +1,179 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_AUTOGEN_DIF_OTP_CTRL_AUTOGEN_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_AUTOGEN_DIF_OTP_CTRL_AUTOGEN_H_
+
+// This file is auto-generated.
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/otp_ctrl/doc/">OTP_CTRL</a> Device Interface Functions
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * A handle to otp_ctrl.
+ *
+ * This type should be treated as opaque by users.
+ */
+typedef struct dif_otp_ctrl {
+  /**
+   * The base address for the otp_ctrl hardware registers.
+   */
+  mmio_region_t base_addr_core;
+  /**
+   * The base address for the otp_ctrl hardware test registers.
+   */
+  mmio_region_t base_addr_prim;
```

Comment:
Little strange to see this here - I dont think any of the CSRs in OTP ctrl need this information. Why add it here?

Reply:
It was in the original DIF. Specifically the `dif_otp_ctrl_read_test(...)` DIF uses it.

Reply:
IMO DIFs are only meant to configure and actuate the HW. These DIFs read or write a chunk of data from the OTP mem region, but they have nothing to do with OTP itself. These could be turned into generic functions where the MMIO region read or written to could be any of the existing memories (OTP, flash, SRAMs, ROM etc). 

I will prefer removing this because it makes it harder to develop auto-generated tests.

Reply:
SG to me. DV is the primary consumer of these so whatever makes it easier for you! After a couple of greps, it looks like there are only two DIFs that read/write from the OTP mem region (`dif_otp_ctrl_write_test` and `dif_otp_ctrl_read_test`) and they are not used anywhere in the source tree. So if there are no objections, I will just deprecate these DIFs and remove the extra base address from the context struct.

Reply:
Fixed.